x-pack/legacy/plugins/monitoring/public/lib/setup_mode.js

```diff
@@ -4,41 +4,42 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React from 'react';
-import { render } from 'react-dom';
 import { ajaxErrorHandlersProvider } from './ajax_error_handler';
 import { get, contains } from 'lodash';
-import chrome from 'ui/chrome';
 import { toastNotifications } from 'ui/notify';
 import { i18n } from '@kbn/i18n';
-import { SetupModeEnterButton } from '../components/setup_mode/enter_button';
 import { npSetup } from 'ui/new_platform';
 
-function isOnPage(hash) {
-  return contains(window.location.hash, hash);
-}
+import {
+  checkInternalMonitoring,
+  destroyInternalMonitoringWarning,
+} from '../components/setup_mode/check_internal_monitoring';
 
 const angularState = {
   injector: null,
   scope: null,
 };
 
-const checkAngularState = () => {
+const isEligible = () => {
```

Comment:
Can you explain the reasoning behind these changes?

Reply:
This function is no longer checking for just the angular state. I decided to throw all the gatekeeping logic into one function to avoid a lot of duplication and "if" conditions. So, the name `checkAngularState` doesn't really make sense anymore

Reply:
Yea I like the idea a lot. I just wonder if we're mixing two different things.

The `checkAngularState` is more of a "hey, you as the dev made a mistake and didn't setup stuff properly" whereas the cloud check is more of a "hey, the code is setup properly but the user isn't in the right environment to see this UI"

Maybe we need two checks - `isReady` (or `isInit`) and `isEligbile`

WDYT?

Reply:
I could break it up, but then we'll just end up doing `isReady && isEligible` anyways. I don't really see them being used separately.

But, I do see your point, so I'll let you make the call on this one 👍

Reply:
Okay let's stick with what you have. Makes sense!